astacus/coordinator/plugins/clickhouse/steps.py

```diff
@@ -435,6 +466,27 @@ async def run_on_every_node(
     await asyncio.gather(*[gather_limited(per_node_concurrency_limit, fn(client)) for client in clients])
 
 
+async def wait_for_condition_on_every_node(
+    clients: Iterable[ClickHouseClient],
+    condition: Callable[[ClickHouseClient], Awaitable[bool]],
+    description: str,
+    timeout_seconds: float,
+    recheck_every_seconds: float = 1.0,
+) -> None:
+    loop = asyncio.get_event_loop()
+
+    async def wait_for_condition(client: ClickHouseClient) -> None:
+        start_time = loop.time()
```

Comment:
It's unclear to me (and from reading the doc), is this time moving forward when another task in running ?

Reply:
I just had a look at the implementation 😆  https://github.com/python/cpython/blob/main/Lib/asyncio/base_events.py#L775

I think I will cut out the middle man